phac-nml/irida-next
# Project workflow executions UI

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR sets up the UI for project workflow executions

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/a4a463cc-6062-44f9-a6fd-26918e61e949)

![image](https://github.com/phac-nml/irida-next/assets/25466211/97460753-fef8-4925-9e4f-686748a98004)

![image](https://github.com/phac-nml/irida-next/assets/25466211/23fa4990-ec66-41c7-b35c-c9df8ed565b4)

![image](https://github.com/phac-nml/irida-next/assets/25466211/dd2d448a-f080-4ac7-ba6f-09f0afa87e9f)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as any user
2. Create a new project
3. Go to the settings for the project and from the expanded menu select `Automated Workflow Executions`
4. Add a new automated workflow execution (this will launch automatically whenever paired-end data is added to a sample in the project)
5. Go to the samples page for the project and create a new sample
6. Upload paired end data to the sample
7. Go to the `Workflow Executions` page for the project
8. You should see the workflow execution has been launched
9. Add a member to the project with less than `Maintainer` access level
10. Login as this user and visit the projects workflow executions page
11. User should not be able to cancel or destroy the automated workflow that was launched
12. Create a new project with this user, add a sample, upload paired-end data
13. Verify no automated workflows have been launched for this project (go to the `Workflow Executions` page for the project
14. Visit the main Workflow Executions page for the current user. You should only see the workflow execution that the user launched
15. Log out and login as the user from step 1
16. Go to the `Workflow Executions` page for the project created in step 2
17. Verify the user can delete or cancel the automated workflow execution

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
